pkg/kube/kube_handler.go

```diff
@@ -415,19 +415,24 @@ func parseRepoRequest(appRepoBody io.ReadCloser) (*appRepositoryRequest, error)
 
 func (a *userHandler) applyAppRepositorySecret(repoSecret *corev1.Secret, requestNamespace string, appRepo *v1alpha1.AppRepository) error {
 	// TODO: pass request context through from user request to clientset.
-	_, err := a.clientset.CoreV1().Secrets(requestNamespace).Create(context.TODO(), repoSecret, metav1.CreateOptions{})
-	if err != nil && k8sErrors.IsAlreadyExists(err) {
-		_, err = a.clientset.CoreV1().Secrets(requestNamespace).Update(context.TODO(), repoSecret, metav1.UpdateOptions{})
-	}
-	if err != nil {
-		return err
+	if _, ok := repoSecret.Data[".dockerconfigjson"]; !ok {
```

Comment:
This also looks equivalent to swallowing an error. Why is this now needed? As in, in what situation is this function, `applyAppRepositorySecret`, called without a `.dockerconfigjson`, where this was not the case previously, and why do we ignore it in that case?

Reply:
this is the function that creates the secret with the auth info. For the `.dockerconfigjson` case it's not necessary because the secret always exists from before.

Reply:
yes, I agree, this `applyAppRepositorySecret` is confusing right now